pgjdbc/src/main/java/org/postgresql/copy/CopyManager.java

```diff
@@ -48,26 +48,24 @@ public CopyManager(BaseConnection connection) throws SQLException {
   }
 
   public CopyIn copyIn(String sql) throws SQLException {
-    CopyOperation op = null;
-    try {
-      op = queryExecutor.startCopy(sql, connection.getAutoCommit());
+    CopyOperation op = queryExecutor.startCopy(sql, connection.getAutoCommit());
+    if (op == null || op instanceof CopyIn) {
```

Comment:
That was the behaviour of the old code. I wanted to keep this PR as a refactoring in the actual sense, not change the behaviour of the code.